libp2p/security/security_multistream.py

```diff
@@ -31,15 +31,14 @@ class SecurityMultistream(ABC):
     multiselect: Multiselect
     multiselect_client: MultiselectClient
 
-    def __init__(
-        self, secure_transports_by_protocol: Mapping[TProtocol, ISecureTransport]
-    ) -> None:
+    def __init__(self, secure_transports_by_protocol: TSecurityOptions = None) -> None:
         self.transports = OrderedDict()
         self.multiselect = Multiselect()
         self.multiselect_client = MultiselectClient()
 
-        for protocol, transport in secure_transports_by_protocol.items():
-            self.add_transport(protocol, transport)
+        if secure_transports_by_protocol is not None:
```

Comment:
nitpick:

```suggestion
        if secure_transports_by_protocol:
```

Reply:
we could also just make this an empty tuple, then in _all_ cases, you just iterate over all of them

Reply:
An empty default value sounds great to me. However, because `secure_transports_by_protocol` is of type `Mapping[TProtocol, ISecureTransport]`, we might still need the pattern for `dict` initialization.
```python
    if not secure_transports_by_protocol:
        secure_transports_by_protocol = {}
```
After tracing the code, I'd just remove the default `None` for `secure_transports_by_protocol` and `muxer_transports_by_protocol` since we didn't use and probably not even need it.